bazel/examples/main.go

```diff
@@ -1,7 +1,12 @@
 package main
 
-import "fmt"
+import (
+	"log"
+	"os"
+)
 
 func main() {
-	fmt.Println("Hello world")
+	if len(os.Args) != 2 || os.Args[1] != "world" {
```

Comment:
I made this change in your branch (something GitHub apparently lets me do for some reason!)

Reply:
In parallel, I split this into two go_binaries: one that expects args, and one that doesn't care.

I also added additional arguments to the arg-checking step, to avoid bugs where our wrapper script only works for exactly 1 arg.

This ended up being a lot easier if I upgraded the go_* bazel rules to match gcs-fetcher's rules, so I did that too.